source/extensions/quic_listeners/quiche/client_connection_factory_impl.h

```diff
@@ -0,0 +1,61 @@
+#pragma once
+
+#include "common/http/http3/quic_client_connection_factory.h"
+#include "common/http/http3/well_known_names.h"
+
+#include "extensions/quic_listeners/quiche/envoy_quic_alarm_factory.h"
+#include "extensions/quic_listeners/quiche/envoy_quic_client_session.h"
+#include "extensions/quic_listeners/quiche/envoy_quic_connection_helper.h"
+#include "extensions/quic_listeners/quiche/envoy_quic_proof_verifier.h"
+#include "extensions/quic_listeners/quiche/envoy_quic_utils.h"
+#include "extensions/transport_sockets/tls/ssl_socket.h"
+
+#include "quiche/quic/core/http/quic_client_push_promise_index.h"
+#include "quiche/quic/core/quic_utils.h"
+
+namespace Envoy {
+namespace Quic {
+
+// TODO(#14829) we should avoid creating this per-connection.
```

Comment:
yeah, I basically thing much of this should be per-pool or per-upstream and if things are per-connection they should actually be set up in the constructor.  Definitely needs cleaning up (tried to make that comment more clear) but I was thinking of doing that and enabling tests in parallel since with the layer of indirection it's going to be messy to have any of the core code create more persistent QUIC structs.  WDYT?

Reply:
SGTM